android/java/org/chromium/chrome/browser/BraveLaunchIntentDispatcher.java

```diff
@@ -0,0 +1,26 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser;
+
+import android.content.Intent;
+import android.content.SharedPreferences;
+
+import org.chromium.base.ContextUtils;
+import org.chromium.chrome.browser.preferences.BravePreferenceKeys;
+
+public class BraveLaunchIntentDispatcher {
```

Comment:
in other classes I left a comment `// see BraveLaunchIntentDispatcherClassAdapter` so it's easier to track down how this works